<?xml version="1.0"?>


<launch>
  <arg name="file_in"   default="benchmark_v2"/>
  <arg name="file_out"  default="mapstrat_out"/>

  <arg name="min_dist"        default="0.10" />
  <arg name="min_time"        default="1" />
  <arg name="time_thresh"     default="100" />
  <arg name="dist_thresh"     default="0.6" />
  <arg name="inliers_thresh"  default="40" />
  <arg name="FOV_thresh"      default="0.33" />
  <arg name="n_kf_local_ba"   default="7" />
  <arg name="freq_global_ba"  default="6" />

  <include file="$(find boris_drone)/launch/global_params.xml" />
  <include file="$(find boris_drone)/launch/gui.xml" />
  <include file="$(find boris_drone)/launch/slam.xml" />

  <node name="boris_drone_manual_pose_estimation" pkg="boris_drone" type="manual_pose_estimation" output="screen"></node>
  <node pkg="rosbag" type="play"   name="player"   output="screen" args="-q /home/$(env USER)/bagfiles/$(arg file_in).bag -r 0.75"/>
  <node name="boris_drone_computer_vision" pkg="boris_drone" type="computer_vision" output="screen">
    <param name="video_channel"        value="ardrone/front/image_rect_color"/>
    <param name="cam_type"             value="front"/>
    <param name="use_OpticalFlowPyrLK" value="true"/>
    <param name="autonomy_unavailable" value="true"/>
  </node>


  <!-- overriding default values of included files -->
  <param name="boris_drone_mapping_node/sonar_unavailable" value="true" />

  <param name="boris_drone_mapping_node/min_dist"               value="$(arg min_dist)" />
  <param name="boris_drone_mapping_node/min_time"               value="$(arg min_time)" />
  <param name="boris_drone_mapping_node/time_thresh"            value="$(arg time_thresh)" />
  <param name="boris_drone_mapping_node/inliers_thresh"         value="$(arg inliers_thresh)" />
  <param name="boris_drone_mapping_node/dist_thresh"            value="$(arg dist_thresh)" />
  <param name="boris_drone_mapping_node/FOV_thresh"             value="$(arg FOV_thresh)" />
  <param name="boris_drone_mapping_node/n_kf_local_ba"          value="$(arg n_kf_local_ba)" />
  <param name="boris_drone_mapping_node/freq_global_ba"         value="$(arg freq_global_ba)" />
  <param name="boris_drone_mapping_node/no_bundle_adjustment"   value="false" />
  <param name="boris_drone_mapping_node/manual_keyframes"       value="false" />
  <param name="boris_drone_mapping_node/sonar_unavailable"      value="true" />
</launch>
